test/test_nullable.cpp

```diff
@@ -12,55 +12,199 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
+#include <tuple>
+
 #include "sparrow/nullable.hpp"
 
 #include "doctest/doctest.h"
 
 namespace sparrow
 {
-    using nullable_double = nullable<double, bool>;
-    using nullable_int = nullable<int, bool>;
+    class Custom
```

Comment:
Could you add a comment clarifying the intended properties of this type? So that we can check it matches the impl.